wordpress-mobile/WordPress-iOS
# Handle stories media uploading post-publish

Fixes #16054

### Related PRs
* [Kanvas](https://github.com/tumblr/kanvas-ios/pull/113)

### Summary

#### Media Uploading

Changes the media upload process for stories to add media to the post through the existing `addMedia` method. Upon saving (and publishing) the post, the media references are updated if upload has completed or observed and then updated once complete.

This prevents the case where media may not be uploaded by the time the Publish button was tapped, leading to a case where post status wasn't changed (due to an issue with media observers clashing).

#### Kanvas Loading States

The Kanvas updates improve the loading state handling for multiple export. The loading indicator should appear during export and hide once export is complete -- about the same time the Prepublishing Sheet is shown.

https://user-images.githubusercontent.com/3250/117711978-fd6b3280-b190-11eb-9ad8-a202ea90ae8d.MOV

## Testing


**Note:** There is a [known issue](https://github.com/tumblr/kanvas-ios/issues/62) with text positioning which will be fixed separately.

* Create a new Story
* Add multiple media items
* Tap the Arrow button in the upper right
* Quickly tap the Publish button without adding a title
* You should see the media uploading in the Drafts list and it should move to Published when complete

## Regression Notes
1. Potential unintended areas of impact

* Any post publishing (both stories and blog posts or pages)
* Any post editing (both stories and blog posts or pages)

2. What I did to test those areas of impact (or what existing automated tests I relied on)

* Tested publishing a new story and editing an existing story
* Tested publishing a new post and editing an existing post (with media)
* Tested publishing a new page and editing an existing page

3. What automated tests I added (or what prevented me from doing so)

I'm not sure where to add automated tests for this.

PR submission checklist:
